src/components/my-url-input.ts

```diff
@@ -1,39 +1,62 @@
 import "@material/mwc-button";
 import "@material/mwc-textfield";
-import { TextField } from "@material/mwc-textfield";
 import { css, CSSResult, html, LitElement, TemplateResult } from "lit";
-import { customElement, state, query, property } from "lit/decorators.js";
-import { DEFAULT_HASS_URL } from "../const";
+import type { TextField } from "@material/mwc-textfield";
+import { customElement, state, property } from "lit/decorators.js";
+import { unsafeSVG } from "lit/directives/unsafe-svg.js";
+import { DEFAULT_HASS_URL, STORAGE_KEY } from "../const";
 import { fireEvent } from "../util/fire_event";
-
-const HASS_URL = "hassUrl";
+import { svgDelete } from "./svg";
 
 @customElement("my-url-input")
 export class MyUrlInputMain extends LitElement {
-  @property() public value?: string;
+  @property() public values?: string[];
 
   @state() private _error?: string | TemplateResult;
 
-  @query("mwc-textfield", true) private _textfield!: TextField;
-
   public focus(): void {
-    this.updateComplete.then(() => this._textfield.focus());
+    this.updateComplete.then(() => this.shadowRoot!.querySelector("mwc-textfield")!.focus());
   }
 
   protected render(): TemplateResult {
     return html`
       ${this._error ? html`<p class="error">${this._error}</p>` : ""}
-      <div>
-        <mwc-textfield
-          label="Home Assistant URL"
-          .value=${this.value || DEFAULT_HASS_URL}
-          @keydown=${this._handleInputKeyDown}
-        ></mwc-textfield>
+      ${(this.values || [DEFAULT_HASS_URL]).map(value => html`
+          <div class="instance-input">
+            <mwc-textfield
+              label="Home Assistant URL"
+              .value=${value}
+              @keydown=${this._handleInputKeyDown}
+            >
+            </mwc-textfield>
+            ${this.values?.length ? html`<mwc-button 
+              .value=${value}
+              @click=${this._handleRemove}>
+              ${unsafeSVG(svgDelete)}
+            </mwc-button>`
+            : html`
+            <mwc-button
+              label="save"
+              @click=${this._handleSave}
+            >
+            </mwc-button>`}
+          </div>
+        `)}
+
+        ${this.values?.length ? html`<div class="buttons">
+      <mwc-button
```

Comment:
I think that this looks better because we separate the finish action from the data mutation action.
![image](https://user-images.githubusercontent.com/1444314/195939689-59e851a4-4733-4eb7-b752-2bce320c937f.png)

Reply:
Note that we should implement it with a `<button type='link'>` for accessibility. (styles can be found in frontend if they are not in this repo yet)